src/instructions/instruction.njk

```diff
@@ -18,22 +18,124 @@ eleventyComputed:
-    </div>
+
+  <div class="bitfield-diagram">
+    <script type="text/bitfield">
+    {{ inst.bitfieldJSON | dump | safe }}
+    </script>
+  </div>
+
+  <script>
+  (function () {
+    function jsonmlToDom(node, inSvg) {
+      if (node === null || node === undefined) return null;
+      if (typeof node === "string" || typeof node === "number") {
+        return document.createTextNode(String(node));
+      }
+      if (!Array.isArray(node) || node.length === 0) return null;
+
+      const [tagName, ...rest] = node;
+      let attrs = {};
+      let children = rest;
+
+      if (children.length && typeof children[0] === "object" && !Array.isArray(children[0])) {
+        attrs = children.shift() || {};
+      }
+
+      const isSvgNode = inSvg || tagName === "svg";
+      const element = isSvgNode
+        ? document.createElementNS("http://www.w3.org/2000/svg", tagName)
+        : document.createElement(tagName);
+
+      Object.entries(attrs).forEach(([key, value]) => {
+        element.setAttribute(key, value);
+      });
+
+      children.forEach(child => {
+        const childNode = jsonmlToDom(child, isSvgNode);
+        if (childNode) element.appendChild(childNode);
+      });
+
+      return element;
+    }
+
+    function parseBitfieldData(raw) {
+      if (!raw) return null;
+      try {
+        const payload = JSON.parse(raw);
+        if (Array.isArray(payload)) {
+          return { segments: payload };
+        }
+        if (payload && Array.isArray(payload.reg)) {
+          return { segments: payload.reg, options: payload.options };
+        }
+        if (payload && Array.isArray(payload.fields)) {
+          return { segments: payload.fields, options: payload.options };
+        }
+      } catch (err) {
+        console.warn("Failed to parse bitfield JSON", err);
+      }
+      return null;
+    }
+
+    function renderBitfieldScripts() {
+      const scripts = document.querySelectorAll('script[type="text/bitfield"]');
+      if (!scripts.length) return;
+
+      scripts.forEach(scriptEl => {
+        const container = scriptEl.parentElement;
+        if (!container) return;
+
+        if (!window.bitfield || typeof window.bitfield.render !== "function") {
+          console.warn("bitfield.render is not available");
+          return;
+        }
+
+        const parsed = parseBitfieldData(scriptEl.textContent.trim());
+        if (!parsed || !Array.isArray(parsed.segments) || !parsed.segments.length) return;
+
+        const segments = parsed.segments
+          .map(seg => {
+            const bits = Number(seg.bits ?? seg.width ?? 0);
+            return Object.assign({}, seg, { bits });
+          })
+          .filter(seg => seg.bits > 0);
+
+        if (!segments.length) return;
+
+        const totalBits = segments.reduce((sum, seg) => sum + seg.bits, 0);
+        const options = Object.assign({ bits: totalBits }, parsed.options || {});
```

Comment:
Similarly to the above, I don't think I quite follow why this logic is necessary. I get that we eventually need to compute the options to pass to `bitfield.render`, but it seems a little weird that those options are being built up in multiple locations: here, in `parseBitFieldData`, and potentially in the thing that generates the JSON. Can we centralize that so that this option computation only appears in one place? That would make it a LOT easier to read/maintain.